// @webstudio-is/sdk-components-react-radix/lib/accordion.js
import { jsx as d } from "react/jsx-runtime";
import { forwardRef as m } from "react";
import { Header as f, Trigger as g, Content as p, Root as A, Item as l } from "@radix-ui/react-accordion";
import { getIndexWithinAncestorFromProps as u } from "@webstudio-is/sdk/runtime";
import { getClosestInstance as a } from "@webstudio-is/react-sdk/runtime";
const $ = m((t, o) => /* @__PURE__ */ d(A, { ref: o, type: "single", ...t })), k = m(({ value: t, ...o }, e) => {
  const n = u(o);
  return /* @__PURE__ */ d(l, { ref: e, value: t ?? n ?? "", ...o });
}), v = f, w = g, y = p, c = "@webstudio-is/sdk-components-react-radix", H = {
  onNavigatorSelect: (t, o) => {
    var e;
    for (const n of o.instancePath)
      if (n.component === `${c}:AccordionContent`) {
        const i = a(
          o.instancePath,
          n,
          `${c}:Accordion`
        ), r = a(
          o.instancePath,
          n,
          `${c}:AccordionItem`
        );
        if (i && r) {
          const s = t.getPropValue(r, "value") ?? ((e = t.indexesWithinAncestors.get(r.id)) == null ? void 0 : e.toString());
          s && t.setMemoryProp(i, "value", s);
        }
      }
  }
};
export {
  $ as Accordion,
  y as AccordionContent,
  v as AccordionHeader,
  k as AccordionItem,
  w as AccordionTrigger,
  H as hooksAccordion
};
