// @webstudio-is/sdk-components-react-radix/lib/accordion.ws.js
import { AccordionIcon as t, ItemIcon as n, HeaderIcon as r, TriggerIcon as c, ContentIcon as i } from "@webstudio-is/icons/svg";
import { defaultStates as a } from "@webstudio-is/sdk";
import { div as o, h3 as p, button as s } from "@webstudio-is/sdk/normalize.css";
import { radix as e } from "./shared/meta.js";
import { buttonReset as d } from "./shared/preset-styles.js";
import { propsAccordion as l, propsAccordionItem as m, propsAccordionHeader as g, propsAccordionTrigger as A, propsAccordionContent as y } from "./__generated__/accordion.props.js";
const x = {
  icon: t,
  contentModel: {
    category: "instance",
    children: ["instance"],
    descendants: [e.AccordionItem]
  },
  presetStyle: { div: o },
  initialProps: ["value", "collapsible"],
  props: l
}, S = {
  label: "Item",
  icon: n,
  indexWithinAncestor: e.Accordion,
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [e.AccordionHeader, e.AccordionContent]
  },
  presetStyle: { div: o },
  initialProps: ["value"],
  props: m
}, C = {
  label: "Item Header",
  icon: r,
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [e.AccordionTrigger]
  },
  presetStyle: {
    h3: [
      ...p,
      {
        property: "margin-top",
        value: { type: "unit", unit: "px", value: 0 }
      },
      {
        property: "margin-bottom",
        value: { type: "unit", unit: "px", value: 0 }
      }
    ]
  },
  props: g
}, H = {
  label: "Item Trigger",
  icon: c,
  contentModel: {
    category: "none",
    children: ["instance", "rich-text"]
  },
  states: [
    ...a,
    {
      category: "component-states",
      label: "Open",
      selector: "[data-state=open]"
    }
  ],
  presetStyle: {
    button: [s, d].flat()
  },
  props: A
}, M = {
  label: "Item Content",
  icon: i,
  contentModel: {
    category: "none",
    children: ["instance", "rich-text"]
  },
  presetStyle: {
    div: o
  },
  props: y
};
export {
  x as metaAccordion,
  M as metaAccordionContent,
  C as metaAccordionHeader,
  S as metaAccordionItem,
  H as metaAccordionTrigger
};
