// @webstudio-is/sdk-components-react-radix/lib/checkbox.ws.js
import { CheckboxCheckedIcon as t, TriggerIcon as o } from "@webstudio-is/icons/svg";
import { defaultStates as e } from "@webstudio-is/sdk";
import { button as c, span as n } from "@webstudio-is/sdk/normalize.css";
import { radix as r } from "./shared/meta.js";
import { buttonReset as a } from "./shared/preset-styles.js";
import { propsCheckbox as s, propsCheckboxIndicator as i } from "./__generated__/checkbox.props.js";
const b = {
  icon: t,
  contentModel: {
    category: "instance",
    children: ["instance"],
    descendants: [r.CheckboxIndicator]
  },
  states: [
    ...e,
    {
      label: "Checked",
      selector: "[data-state=checked]",
      category: "component-states"
    },
    {
      label: "Unchecked",
      selector: "[data-state=unchecked]",
      category: "component-states"
    }
  ],
  presetStyle: {
    button: [c, a].flat()
  },
  initialProps: ["id", "class", "name", "value", "required", "checked"],
  props: s
}, x = {
  icon: o,
  contentModel: {
    category: "none",
    children: ["instance", "rich-text"]
  },
  states: e,
  presetStyle: {
    span: n
  },
  props: i
};
export {
  b as metaCheckbox,
  x as metaCheckboxIndicator
};
