// @webstudio-is/sdk-components-react-radix/lib/checkbox.template.js
import { jsxs as c, jsx as e } from "react/jsx-runtime";
import { CheckMarkIcon as a } from "@webstudio-is/icons/svg";
import { css as o, $ as t, PlaceholderValue as l } from "@webstudio-is/template";
import { radix as r } from "./shared/proxy.js";
import { spacing as s, height as d, width as n, borderRadius as h, borderWidth as b, colors as i, boxShadow as m, opacity as p } from "./shared/theme.js";
const y = {
  category: "radix",
  description: "Use within a form to allow your users to toggle between checked and not checked. Group checkboxes by matching their “Name” properties. Unlike radios, any number of checkboxes in a group can be checked.",
  order: 101,
  template: /* @__PURE__ */ c(
    r.Label,
    {
      "ws:label": "Checkbox Field",
      "ws:style": o`
        display: flex;
        gap: ${s[2]};
        align-items: center;
      `,
      children: [
        /* @__PURE__ */ e(
          r.Checkbox,
          {
            "ws:style": o`
          height: ${d[4]};
          width: ${n[4]};
          flex-shrink: 0;
          border-radius: ${h.sm};
          border: ${b.DEFAULT} solid ${i.primary};
          &:focus-visible {
            outline: none;
            box-shadow: ${m.ring};
          }
          &:disabled {
            cursor: not-allowed;
            opacity: ${p[50]};
          }
          &[data-state="checked"] {
            background-color: ${i.primary};
            color: ${i.primaryForeground};
          }
        `,
            children: /* @__PURE__ */ e(
              r.CheckboxIndicator,
              {
                "ws:style": o`
            display: flex;
            align-items: center;
            justify-content: center;
            color: currentColor;
          `,
                children: /* @__PURE__ */ e(t.HtmlEmbed, { "ws:label": "Indicator Icon", code: a })
              }
            )
          }
        ),
        /* @__PURE__ */ e(t.Text, { "ws:label": "Checkbox Label", "ws:tag": "span", children: new l("Checkbox") })
      ]
    }
  )
};
export {
  y as meta
};
