// @webstudio-is/sdk-components-react-radix/lib/collapsible.ws.js
import { CollapsibleIcon as n, TriggerIcon as t, ContentIcon as l } from "@webstudio-is/icons/svg";
import { div as o } from "@webstudio-is/sdk/normalize.css";
import { radix as e } from "./shared/meta.js";
import { propsCollapsible as r, propsCollapsibleTrigger as p, propsCollapsibleContent as i } from "./__generated__/collapsible.props.js";
const g = {
  icon: n,
  contentModel: {
    category: "instance",
    children: ["instance"],
    descendants: [e.CollapsibleTrigger, e.CollapsibleContent]
  },
  presetStyle: {
    div: o
  }
}, b = {
  icon: t,
  contentModel: {
    category: "none",
    children: ["instance", "rich-text"]
  }
}, d = {
  icon: l,
  contentModel: {
    category: "none",
    children: ["instance", "rich-text"]
  },
  presetStyle: {
    div: o
  }
}, m = {
  props: {
    ...r,
    onOpenChange: {
      type: "action",
      control: "action",
      required: !1
    }
  },
  initialProps: ["open", "onOpenChange"]
}, h = {
  props: p
}, y = {
  props: i
};
export {
  g as metaCollapsible,
  d as metaCollapsibleContent,
  b as metaCollapsibleTrigger,
  m as propsMetaCollapsible,
  y as propsMetaCollapsibleContent,
  h as propsMetaCollapsibleTrigger
};
