// @webstudio-is/sdk-components-react-radix/lib/dialog.ws.js
import { DialogIcon as t, TriggerIcon as n, OverlayIcon as i, ContentIcon as r, ButtonElementIcon as c, HeadingIcon as a, TextIcon as l } from "@webstudio-is/icons/svg";
import { defaultStates as s } from "@webstudio-is/sdk";
import { div as e, button as p, h2 as g, p as d } from "@webstudio-is/sdk/normalize.css";
import { radix as o } from "./shared/meta.js";
import { propsDialog as D, propsDialogTrigger as m, propsDialogOverlay as y, propsDialogContent as h, propsDialogClose as f, propsDialogTitle as C, propsDialogDescription as T } from "./__generated__/dialog.props.js";
import { buttonReset as I } from "./shared/preset-styles.js";
const O = {
  icon: n,
  contentModel: {
    category: "none",
    children: ["instance"]
  },
  props: m
}, B = {
  icon: i,
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [o.DialogContent]
  },
  presetStyle: { div: e },
  props: y
}, E = {
  icon: r,
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [
      o.DialogTitle,
      o.DialogDescription,
      o.DialogClose
    ]
  },
  presetStyle: { div: e },
  props: h
}, H = {
  icon: a,
  contentModel: {
    category: "none",
    children: ["instance", "rich-text"]
  },
  presetStyle: { h2: g },
  props: C
}, R = {
  icon: l,
  contentModel: {
    category: "none",
    children: ["instance", "rich-text"]
  },
  presetStyle: { p: d },
  props: T
}, j = {
  icon: c,
  label: "Close Button",
  contentModel: {
    category: "none",
    children: ["instance", "rich-text"]
  },
  states: s,
  presetStyle: {
    button: [I, p].flat()
  },
  props: f
}, k = {
  icon: t,
  contentModel: {
    category: "instance",
    children: ["instance"],
    descendants: [o.DialogTrigger, o.DialogOverlay]
  },
  props: D
};
export {
  k as metaDialog,
  j as metaDialogClose,
  E as metaDialogContent,
  R as metaDialogDescription,
  B as metaDialogOverlay,
  H as metaDialogTitle,
  O as metaDialogTrigger
};
