// @webstudio-is/sdk-components-react-radix/lib/navigation-menu.ws.js
import { NavigationMenuIcon as t, ListIcon as o, ListItemIcon as i, TriggerIcon as a, ContentIcon as r, BoxIcon as s, ViewportIcon as c } from "@webstudio-is/icons/svg";
import { div as e } from "@webstudio-is/sdk/normalize.css";
import { radix as n } from "./shared/meta.js";
import { propsNavigationMenu as p, propsNavigationMenuItem as g, propsNavigationMenuTrigger as M, propsNavigationMenuContent as u, propsNavigationMenuLink as l, propsNavigationMenuList as v, propsNavigationMenuViewport as N } from "./__generated__/navigation-menu.props.js";
const y = {
  icon: t,
  contentModel: {
    category: "instance",
    children: ["instance"],
    descendants: [n.NavigationMenuList, n.NavigationMenuViewport]
  },
  presetStyle: {
    div: e
  }
}, h = {
  icon: o,
  label: "Menu List",
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [n.NavigationMenuItem]
  },
  presetStyle: {
    div: e
  }
}, b = {
  icon: i,
  label: "Menu Item",
  indexWithinAncestor: n.NavigationMenu,
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [
      n.NavigationMenuTrigger,
      n.NavigationMenuContent,
      n.NavigationMenuLink
    ]
  },
  presetStyle: {
    div: e
  }
}, k = {
  icon: a,
  label: "Menu Trigger",
  contentModel: {
    category: "none",
    children: ["instance"]
  }
}, w = {
  icon: r,
  label: "Menu Content",
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [n.NavigationMenuLink]
  },
  presetStyle: {
    div: e
  }
}, C = {
  icon: s,
  label: "Accessible Link Wrapper",
  contentModel: {
    category: "none",
    children: ["instance"]
  }
}, T = {
  icon: c,
  label: "Menu Viewport",
  contentModel: {
    category: "none",
    children: ["instance"]
  },
  presetStyle: {
    div: e
  }
}, V = {
  props: p
}, S = {
  props: g
}, f = {
  props: M
}, x = {
  props: u
}, A = {
  props: l
}, W = {
  props: v
}, B = {
  props: N
};
export {
  y as metaNavigationMenu,
  w as metaNavigationMenuContent,
  b as metaNavigationMenuItem,
  C as metaNavigationMenuLink,
  h as metaNavigationMenuList,
  k as metaNavigationMenuTrigger,
  T as metaNavigationMenuViewport,
  V as propsMetaNavigationMenu,
  x as propsMetaNavigationMenuContent,
  S as propsMetaNavigationMenuItem,
  A as propsMetaNavigationMenuLink,
  W as propsMetaNavigationMenuList,
  f as propsMetaNavigationMenuTrigger,
  B as propsMetaNavigationMenuViewport
};
