// @webstudio-is/sdk-components-react-radix/lib/popover.template.js
import { jsxs as i, jsx as o } from "react/jsx-runtime";
import { $ as t, PlaceholderValue as d, css as n } from "@webstudio-is/template";
import { radix as r } from "./shared/proxy.js";
import { getButtonStyle as l } from "./shared/styles.js";
import { zIndex as a, width as p, borderRadius as s, borderWidth as c, colors as e, spacing as m, boxShadow as h } from "./shared/theme.js";
const w = {
  category: "radix",
  description: "Displays rich content in a portal, triggered by a button.",
  order: 6,
  template: /* @__PURE__ */ i(r.Popover, { children: [
    /* @__PURE__ */ o(r.PopoverTrigger, { children: /* @__PURE__ */ o(t.Button, { "ws:style": l("outline"), children: new d("Button") }) }),
    /* @__PURE__ */ o(
      r.PopoverContent,
      {
        "ws:style": n`
          z-index: ${a[50]};
          width: ${p[72]};
          border-radius: ${s.md};
          border: ${c.DEFAULT} solid ${e.border};
          background-color: ${e.popover};
          padding: ${m[4]};
          color: ${e.popoverForeground};
          box-shadow: ${h.md};
          outline: none;
        `,
        children: /* @__PURE__ */ o(t.Text, { children: new d("The text you can edit") })
      }
    )
  ] })
};
export {
  w as meta
};
