// @webstudio-is/sdk-components-react-radix/lib/popover.ws.js
import { PopoverIcon as e, TriggerIcon as t, ContentIcon as n, ButtonElementIcon as r } from "@webstudio-is/icons/svg";
import { defaultStates as p } from "@webstudio-is/sdk";
import { div as s, button as i } from "@webstudio-is/sdk/normalize.css";
import { radix as o } from "./shared/meta.js";
import { propsPopover as c, propsPopoverTrigger as a, propsPopoverContent as l, propsPopoverClose as d } from "./__generated__/popover.props.js";
import { buttonReset as m } from "./shared/preset-styles.js";
const y = {
  icon: t,
  contentModel: {
    category: "none",
    children: ["instance"]
  },
  props: a
}, h = {
  icon: n,
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [o.PopoverClose]
  },
  presetStyle: {
    div: s
  },
  initialProps: ["side", "sideOffset", "align", "alignOffset"],
  props: l
}, b = {
  icon: e,
  contentModel: {
    category: "instance",
    children: ["instance"],
    descendants: [o.PopoverTrigger, o.PopoverContent]
  },
  initialProps: ["open"],
  props: c
}, I = {
  icon: r,
  label: "Close Button",
  contentModel: {
    category: "none",
    children: ["instance", "rich-text"]
  },
  states: p,
  presetStyle: {
    button: [m, i].flat()
  },
  props: d
};
export {
  b as metaPopover,
  I as metaPopoverClose,
  h as metaPopoverContent,
  y as metaPopoverTrigger
};
