// @webstudio-is/sdk-components-react-radix/lib/popover.ws.js
import { TriggerIcon as e, ContentIcon as n, PopoverIcon as r } from "@webstudio-is/icons/svg";
import { div as t } from "@webstudio-is/sdk/normalize.css";
import { radix as o } from "./shared/meta.js";
import { propsPopover as p, propsPopoverTrigger as i, propsPopoverContent as s } from "./__generated__/popover.props.js";
const v = {
  icon: e,
  contentModel: {
    category: "none",
    children: ["instance"]
  }
}, d = {
  icon: n,
  contentModel: {
    category: "none",
    children: ["instance"]
  },
  presetStyle: {
    div: t
  }
}, l = {
  icon: r,
  contentModel: {
    category: "instance",
    children: ["instance"],
    descendants: [o.PopoverTrigger, o.PopoverContent]
  }
}, m = {
  props: p,
  initialProps: ["open"]
}, f = {
  props: i
}, M = {
  props: s,
  initialProps: ["side", "sideOffset", "align", "alignOffset"]
};
export {
  l as metaPopover,
  d as metaPopoverContent,
  v as metaPopoverTrigger,
  m as propsMetaPopover,
  M as propsMetaPopoverContent,
  f as propsMetaPopoverTrigger
};
