// @webstudio-is/sdk-components-react-radix/lib/radio-group.ws.js
import { RadioGroupIcon as o, ItemIcon as r, TriggerIcon as a } from "@webstudio-is/icons/svg";
import { defaultStates as t } from "@webstudio-is/sdk";
import { div as n, button as c, span as s } from "@webstudio-is/sdk/normalize.css";
import { radix as e } from "./shared/meta.js";
import { buttonReset as i } from "./shared/preset-styles.js";
import { propsRadioGroup as p, propsRadioGroupItem as d, propsRadioGroupIndicator as l } from "./__generated__/radio-group.props.js";
const y = {
  icon: o,
  contentModel: {
    category: "instance",
    children: ["instance"],
    descendants: [e.RadioGroupItem]
  },
  states: [
    ...t,
    {
      label: "Checked",
      selector: "[data-state=checked]",
      category: "component-states"
    },
    {
      label: "Unchecked",
      selector: "[data-state=unchecked]",
      category: "component-states"
    }
  ],
  presetStyle: {
    div: n
  },
  initialProps: ["id", "class", "name", "value", "required"],
  props: p
}, g = {
  icon: r,
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [e.RadioGroupIndicator]
  },
  states: t,
  presetStyle: {
    button: [c, i].flat()
  },
  initialProps: ["value"],
  props: d
}, h = {
  icon: a,
  contentModel: {
    category: "none",
    children: ["instance"]
  },
  states: t,
  presetStyle: {
    span: s
  },
  props: l
};
export {
  y as metaRadioGroup,
  h as metaRadioGroupIndicator,
  g as metaRadioGroupItem
};
