// @webstudio-is/sdk-components-react-radix/lib/radio-group.ws.js
import { RadioGroupIcon as e, ItemIcon as a, TriggerIcon as r } from "@webstudio-is/icons/svg";
import { defaultStates as o } from "@webstudio-is/sdk";
import { div as n, button as s, span as c } from "@webstudio-is/sdk/normalize.css";
import { radix as t } from "./shared/meta.js";
import { buttonReset as p } from "./shared/preset-styles.js";
import { propsRadioGroup as i, propsRadioGroupItem as d, propsRadioGroupIndicator as m } from "./__generated__/radio-group.props.js";
const y = {
  icon: e,
  contentModel: {
    category: "instance",
    children: ["instance"],
    descendants: [t.RadioGroupItem]
  },
  states: [
    ...o,
    {
      label: "Checked",
      selector: "[data-state=checked]",
      category: "component-states"
    },
    {
      label: "Unchecked",
      selector: "[data-state=unchecked]",
      category: "component-states"
    }
  ],
  presetStyle: {
    div: n
  }
}, g = {
  icon: a,
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [t.RadioGroupIndicator]
  },
  states: o,
  presetStyle: {
    button: [s, p].flat()
  }
}, h = {
  icon: r,
  contentModel: {
    category: "none",
    children: ["instance"]
  },
  states: o,
  presetStyle: {
    span: c
  }
}, M = {
  props: i,
  initialProps: ["id", "className", "name", "value", "required"]
}, b = {
  props: d,
  initialProps: ["value"]
}, k = {
  props: m
};
export {
  y as metaRadioGroup,
  h as metaRadioGroupIndicator,
  g as metaRadioGroupItem,
  M as propsMetaRadioGroup,
  k as propsMetaRadioGroupIndicator,
  b as propsMetaRadioGroupItem
};
