// @webstudio-is/sdk-components-react-radix/lib/select.ws.js
import { SelectIcon as o, TriggerIcon as c, FormTextFieldIcon as r, ContentIcon as l, ViewportIcon as p, ItemIcon as s, CheckMarkIcon as a, TextIcon as i } from "@webstudio-is/icons/svg";
import { button as S, span as t, div as n } from "@webstudio-is/sdk/normalize.css";
import { radix as e } from "./shared/meta.js";
import { propsSelect as d, propsSelectTrigger as m, propsSelectValue as I, propsSelectContent as g, propsSelectViewport as M, propsSelectItem as y, propsSelectItemIndicator as T, propsSelectItemText as h } from "./__generated__/select.props.js";
const w = {
  icon: o,
  contentModel: {
    category: "instance",
    children: ["instance"],
    descendants: [e.SelectTrigger, e.SelectContent]
  }
}, b = {
  icon: c,
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [e.SelectValue]
  },
  presetStyle: {
    button: S
  }
}, f = {
  label: "Value",
  icon: r,
  contentModel: {
    category: "none",
    children: []
  },
  presetStyle: {
    span: t
  }
}, v = {
  icon: l,
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [e.SelectViewport]
  },
  presetStyle: {
    div: n
  }
}, P = {
  icon: p,
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [e.SelectItem]
  },
  presetStyle: {
    div: n
  }
}, k = {
  icon: s,
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [e.SelectItemIndicator, e.SelectItemText]
  },
  presetStyle: {
    div: n
  }
}, F = {
  label: "Indicator",
  icon: a,
  contentModel: {
    category: "none",
    children: ["instance"]
  },
  presetStyle: {
    span: t
  }
}, q = {
  label: "Item Text",
  icon: i,
  contentModel: {
    category: "none",
    children: ["instance", "rich-text"]
  },
  presetStyle: {
    span: t
  }
}, j = {
  props: d,
  initialProps: ["name", "value", "open", "required"]
}, z = {
  props: m
}, A = {
  props: I,
  initialProps: ["placeholder"]
}, B = {
  props: g
}, D = {
  props: M
}, E = {
  props: y,
  initialProps: ["value"]
}, G = {
  props: T
}, H = {
  props: h
};
export {
  w as metaSelect,
  v as metaSelectContent,
  k as metaSelectItem,
  F as metaSelectItemIndicator,
  q as metaSelectItemText,
  b as metaSelectTrigger,
  f as metaSelectValue,
  P as metaSelectViewport,
  j as propsMetaSelect,
  B as propsMetaSelectContent,
  E as propsMetaSelectItem,
  G as propsMetaSelectItemIndicator,
  H as propsMetaSelectItemText,
  z as propsMetaSelectTrigger,
  A as propsMetaSelectValue,
  D as propsMetaSelectViewport
};
