// @webstudio-is/sdk-components-react-radix/lib/switch.template.js
import { jsx as o } from "react/jsx-runtime";
import { css as r } from "@webstudio-is/template";
import { radix as e } from "./shared/proxy.js";
import { borderRadius as a, borderWidth as d, transition as i, boxShadow as n, opacity as s, colors as t, height as c, width as l } from "./shared/theme.js";
const m = {
  category: "radix",
  description: "A control that allows the user to toggle between checked and not checked.",
  order: 11,
  template: /* @__PURE__ */ o(
    e.Switch,
    {
      "ws:style": r`
        display: inline-flex;
        height: 24px;
        width: 44px;
        flex-shrink: 0;
        cursor: pointer;
        align-items: center;
        border-radius: ${a.full};
        border: ${d[2]} solid transparent;
        transition: ${i.all};
        &:focus-visible {
          outline: none;
          box-shadow: ${n.ring};
        }
        &:disabled {
          cursor: not-allowed;
          opacity: ${s[50]};
        }
        &[data-state="checked"] {
          background-color: ${t.primary};
        }
        &[data-state="unchecked"] {
          background-color: ${t.input};
        }
      `,
      children: /* @__PURE__ */ o(
        e.SwitchThumb,
        {
          "ws:style": r`
          pointer-events: none;
          display: block;
          height: ${c[5]};
          width: ${l[5]};
          border-radius: ${a.full};
          background-color: ${t.background};
          box-shadow: ${n.lg};
          transition: ${i.transform};
          &[data-state="checked"] {
            transform: translateX(20px);
          }
          &[data-state="unchecked"] {
            transform: translateX(0px);
          }
        `
        }
      )
    }
  )
};
export {
  m as meta
};
