// @webstudio-is/sdk-components-react-radix/lib/switch.ws.js
import { SwitchIcon as e, TriggerIcon as c } from "@webstudio-is/icons/svg";
import { defaultStates as t } from "@webstudio-is/sdk";
import { button as o, span as a } from "@webstudio-is/sdk/normalize.css";
import { radix as s } from "./shared/meta.js";
import { buttonReset as n } from "./shared/preset-styles.js";
import { propsSwitch as r, propsSwitchThumb as i } from "./__generated__/switch.props.js";
const b = {
  icon: e,
  contentModel: {
    category: "instance",
    children: ["instance"],
    descendants: [s.SwitchThumb]
  },
  states: [
    ...t,
    {
      label: "Checked",
      selector: "[data-state=checked]",
      category: "component-states"
    },
    {
      label: "Unchecked",
      selector: "[data-state=unchecked]",
      category: "component-states"
    }
  ],
  presetStyle: {
    button: [o, n].flat()
  }
}, S = {
  icon: c,
  contentModel: {
    category: "none",
    children: ["instance"]
  },
  states: [
    ...t,
    {
      label: "Checked",
      selector: "[data-state=checked]",
      category: "component-states"
    },
    {
      label: "Unchecked",
      selector: "[data-state=unchecked]",
      category: "component-states"
    }
  ],
  presetStyle: {
    span: a
  }
}, k = {
  props: r,
  initialProps: ["id", "className", "name", "value", "checked", "required"]
}, f = {
  props: i
};
export {
  b as metaSwitch,
  S as metaSwitchThumb,
  k as propsMetaSwitch,
  f as propsMetaSwitchThumb
};
