// @webstudio-is/sdk-components-react-radix/lib/switch.ws.js
import { SwitchIcon as t, TriggerIcon as c } from "@webstudio-is/icons/svg";
import { defaultStates as e } from "@webstudio-is/sdk";
import { button as o, span as a } from "@webstudio-is/sdk/normalize.css";
import { radix as n } from "./shared/meta.js";
import { buttonReset as s } from "./shared/preset-styles.js";
import { propsSwitch as r, propsSwitchThumb as i } from "./__generated__/switch.props.js";
const b = {
  icon: t,
  contentModel: {
    category: "instance",
    children: ["instance"],
    descendants: [n.SwitchThumb]
  },
  states: [
    ...e,
    {
      label: "Checked",
      selector: "[data-state=checked]",
      category: "component-states"
    },
    {
      label: "Unchecked",
      selector: "[data-state=unchecked]",
      category: "component-states"
    }
  ],
  presetStyle: {
    button: [o, s].flat()
  },
  initialProps: ["id", "class", "name", "value", "checked", "required"],
  props: r
}, k = {
  icon: c,
  contentModel: {
    category: "none",
    children: ["instance"]
  },
  states: [
    ...e,
    {
      label: "Checked",
      selector: "[data-state=checked]",
      category: "component-states"
    },
    {
      label: "Unchecked",
      selector: "[data-state=unchecked]",
      category: "component-states"
    }
  ],
  presetStyle: {
    span: a
  },
  props: i
};
export {
  b as metaSwitch,
  k as metaSwitchThumb
};
