// @webstudio-is/sdk-components-react-radix/lib/tabs.ws.js
import { TabsIcon as n, HeaderIcon as o, TriggerIcon as s, ContentIcon as r } from "@webstudio-is/icons/svg";
import { defaultStates as a } from "@webstudio-is/sdk";
import { div as e, button as c } from "@webstudio-is/sdk/normalize.css";
import { radix as t } from "./shared/meta.js";
import { buttonReset as i } from "./shared/preset-styles.js";
import { propsTabs as p, propsTabsList as b, propsTabsTrigger as T, propsTabsContent as l } from "./__generated__/tabs.props.js";
const M = {
  icon: n,
  contentModel: {
    category: "instance",
    children: ["instance"],
    descendants: [t.TabsList, t.TabsContent]
  },
  presetStyle: {
    div: e
  }
}, x = {
  icon: o,
  contentModel: {
    category: "none",
    children: ["instance"],
    descendants: [t.TabsTrigger]
  },
  presetStyle: {
    div: e
  }
}, C = {
  icon: s,
  label: "Tab Trigger",
  indexWithinAncestor: t.Tabs,
  contentModel: {
    category: "none",
    children: ["instance", "rich-text"]
  },
  states: [
    ...a,
    {
      category: "component-states",
      label: "Active",
      selector: "[data-state=active]"
    }
  ],
  presetStyle: {
    button: [c, i].flat()
  }
}, S = {
  label: "Tab Content",
  icon: r,
  indexWithinAncestor: t.Tabs,
  contentModel: {
    category: "none",
    children: ["instance", "rich-text"]
  },
  presetStyle: {
    div: e
  }
}, u = {
  props: p
}, I = {
  props: b
}, L = {
  props: T
}, v = {
  props: l
};
export {
  M as metaTabs,
  S as metaTabsContent,
  x as metaTabsList,
  C as metaTabsTrigger,
  u as propsMetaTabs,
  v as propsMetaTabsContent,
  I as propsMetaTabsList,
  L as propsMetaTabsTrigger
};
