// @webstudio-is/sdk-components-react-radix/lib/tabs.template.js
import { jsxs as s, jsx as t } from "react/jsx-runtime";
import { css as a, PlaceholderValue as o } from "@webstudio-is/template";
import { radix as e } from "./shared/proxy.js";
import { borderRadius as l, spacing as n, fontSize as g, fontSizeLineHeight as h, weights as u, transition as b, boxShadow as r, opacity as p, colors as i, height as m } from "./shared/theme.js";
const d = a`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: ${l.md};
  padding: ${n[1.5]} ${n[3]};
  font-size: ${g.sm};
  line-height: ${h.sm};
  font-weight: ${u.medium};
  transition: ${b.all};
  &:focus-visible {
    outline: 2px solid transparent;
    outline-offset: 2px;
    box-shadow: ${r.ring};
  }
  &:disabled {
    pointer-events: none;
    opacity: ${p[50]};
  }
  &[data-state="active"] {
    background-color: ${i.background};
    color: ${i.foreground};
    box-shadow: ${r.sm};
  }
`, c = a`
  margin-top: ${n[2]};
  &:focus-visible {
    outline: none;
    box-shadow: ${r.ring};
  }
`, x = {
  category: "radix",
  description: "A set of panels with content that are displayed one at a time. Duplicate both a tab trigger and tab content to add more tabs. Triggers and content are connected according to their order in the Navigator.",
  order: 2,
  template: /* @__PURE__ */ s(e.Tabs, { defaultValue: "0", children: [
    /* @__PURE__ */ s(
      e.TabsList,
      {
        "ws:style": a`
          display: inline-flex;
          height: ${m[10]};
          align-items: center;
          justify-content: center;
          border-radius: ${l.md};
          background-color: ${i.muted};
          padding: ${n[1]};
          color: ${i.mutedForeground};
        `,
        children: [
          /* @__PURE__ */ t(e.TabsTrigger, { "ws:style": d, children: new o("Account") }),
          /* @__PURE__ */ t(e.TabsTrigger, { "ws:style": d, children: new o("Password") })
        ]
      }
    ),
    /* @__PURE__ */ t(e.TabsContent, { "ws:style": c, children: new o("Make changes to your account here.") }),
    /* @__PURE__ */ t(e.TabsContent, { "ws:style": c, children: new o("Change your password here.") })
  ] })
};
export {
  x as meta
};
