// @webstudio-is/sdk-components-react/lib/__generated__/checkbox.props.js
const e = {
  about: { required: !1, control: "text", type: "string" },
  accept: {
    required: !1,
    control: "text",
    type: "string",
    description: "List of types the server accepts, typically a file type."
  },
  accessKey: {
    required: !1,
    control: "text",
    type: "string",
    description: "Keyboard shortcut to activate or add focus to the element."
  },
  alt: {
    required: !1,
    control: "text",
    type: "string",
    description: "Text description of the image, which is very important for accessibility and search engine optimization. Screen readers read this description to users so they know what the image means. Alt text is also displayed on the page if the image can't be loaded for some reason."
  },
  autoCapitalize: {
    required: !1,
    control: "text",
    type: "string",
    description: "Sets whether input is automatically capitalized when entered by user."
  },
  autoCorrect: { required: !1, control: "text", type: "string" },
  autoFocus: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates that an element should be focused on page load, or when its parent dialog is displayed."
  },
  autoSave: { required: !1, control: "text", type: "string" },
  checked: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates whether the element should be checked on page load."
  },
  className: { required: !1, control: "text", type: "string" },
  color: {
    required: !1,
    control: "color",
    type: "string",
    description: "This attribute sets the text color using either a named color or a  color specified in the hexadecimal #RRGGBB format. Note: This is a legacy attribute. Please use the CSS color property instead."
  },
  content: {
    required: !1,
    control: "text",
    type: "string",
    description: "A value associated with http-equiv orname depending on the context."
  },
  contextMenu: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines the ID of a menu element which willserve as the element's context menu."
  },
  datatype: { required: !1, control: "text", type: "string" },
  defaultChecked: { required: !1, control: "boolean", type: "boolean" },
  defaultValue: { required: !1, control: "text", type: "string" },
  dir: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines the text direction. Allowed values are ltr (Left-To-Right) or rtl (Right-To-Left)"
  },
  disabled: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates whether the user can interact with the element."
  },
  draggable: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Defines whether the element can be dragged."
  },
  enterKeyHint: {
    required: !1,
    control: "select",
    type: "string",
    options: ["search", "enter", "done", "go", "next", "previous", "send"],
    description: "The enterkeyhint specifies what action label (or icon) to present for the enter key onvirtual keyboards. The attribute can be used with form controls (such asthe value of textarea elements), or in elements in anediting host (e.g., using contenteditable attribute)."
  },
  form: {
    required: !1,
    control: "text",
    type: "string",
    description: "Indicates the form that is the owner of the element."
  },
  formAction: {
    required: !1,
    control: "text",
    type: "string",
    description: "Indicates the action of the element, overriding the action defined inthe form."
  },
  formEncType: {
    required: !1,
    control: "text",
    type: "string",
    description: `If the button/input is a submit button (e.g. type="submit"), this attribute sets the encoding type to use during form submission. If this attribute is specified, it overrides theenctype attribute of the button's form owner.`
  },
  formMethod: {
    required: !1,
    control: "text",
    type: "string",
    description: `If the button/input is a submit button (e.g. type="submit"), this attribute sets the submission method to use during form submission (GET, POST, etc.). If this attribute is specified, it overrides the method attribute of the button's form owner.`
  },
  formNoValidate: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: `If the button/input is a submit button (e.g. type="submit"), this boolean attribute specifies that the form is not to be validatedwhen it is submitted. If this attribute is specified, it overrides thenovalidate attribute of the button's form owner.`
  },
  formTarget: {
    required: !1,
    control: "text",
    type: "string",
    description: `If the button/input is a submit button (e.g. type="submit"), this attribute specifies the browsing context (for example, tab, window, or inline frame) in which to display the response that is received aftersubmitting the form. If this attribute is specified, it overrides thetarget attribute of the button's form owner.`
  },
  height: {
    required: !1,
    control: "number",
    type: "number",
    description: "Defines the image’s height in pixels."
  },
  hidden: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Prevents rendering of given element, while keeping child elements, e.g. script elements, active."
  },
  id: {
    required: !1,
    control: "text",
    type: "string",
    description: "Often used with CSS to style a specific element. The value of this attribute must be unique."
  },
  inputMode: {
    description: "Hints at the type of data that might be entered by the user while editing the element or its contents",
    required: !1,
    control: "select",
    type: "string",
    options: [
      "search",
      "text",
      "email",
      "tel",
      "none",
      "url",
      "numeric",
      "decimal"
    ]
  },
  is: {
    description: "Specify that a standard HTML element should behave like a defined custom built-in element",
    required: !1,
    control: "text",
    type: "string"
  },
  itemID: { required: !1, control: "text", type: "string" },
  itemProp: { required: !1, control: "text", type: "string" },
  itemRef: { required: !1, control: "text", type: "string" },
  itemScope: { required: !1, control: "boolean", type: "boolean" },
  itemType: { required: !1, control: "text", type: "string" },
  lang: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines the language used in the element."
  },
  list: {
    required: !1,
    control: "text",
    type: "string",
    description: "Identifies a list of pre-defined options to suggest to the user."
  },
  max: {
    required: !1,
    control: "number",
    type: "number",
    description: "Indicates the maximum value allowed."
  },
  maxLength: {
    required: !1,
    control: "number",
    type: "number",
    description: "Defines the maximum number of characters allowed in the element."
  },
  min: {
    required: !1,
    control: "number",
    type: "number",
    description: "Indicates the minimum value allowed."
  },
  minLength: {
    required: !1,
    control: "number",
    type: "number",
    description: "Defines the minimum number of characters allowed in the element."
  },
  multiple: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates whether multiple values can be entered in an input of the type email or file."
  },
  name: {
    required: !1,
    control: "text",
    type: "string",
    description: "This name is important when submitting form data to the server, as it identifies the data associated with the input. When multiple inputs share the same name attribute, they are treated as part of the same group (e.g., radio buttons or checkboxes)."
  },
  nonce: { required: !1, control: "text", type: "string" },
  pattern: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines a regular expression which the element's value will be validated against."
  },
  placeholder: {
    required: !1,
    control: "text",
    type: "string",
    description: "Provides a hint to the user of what can be entered in the field."
  },
  prefix: { required: !1, control: "text", type: "string" },
  property: { required: !1, control: "text", type: "string" },
  radioGroup: { required: !1, control: "text", type: "string" },
  readOnly: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates whether the element can be edited."
  },
  rel: {
    required: !1,
    control: "text",
    type: "string",
    description: "Specifies the relationship of the target object to the link object."
  },
  required: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates whether this form element must be filled before the form can be submitted."
  },
  resource: { required: !1, control: "text", type: "string" },
  results: { required: !1, control: "number", type: "number" },
  rev: { required: !1, control: "text", type: "string" },
  role: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines an explicit role for an element for use by assistive technologies."
  },
  security: { required: !1, control: "text", type: "string" },
  size: {
    required: !1,
    control: "number",
    type: "number",
    description: "Defines the width of the element (in pixels). If the element'stype attribute is text or password then it's the number of characters."
  },
  slot: {
    required: !1,
    control: "text",
    type: "string",
    description: "Assigns a slot in a shadow DOM shadow tree to an element."
  },
  spellCheck: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates whether spell checking is allowed for the element."
  },
  src: {
    required: !1,
    control: "text",
    type: "string",
    description: "The URL of the embeddable content."
  },
  step: { required: !1, control: "number", type: "number" },
  suppressContentEditableWarning: {
    required: !1,
    control: "boolean",
    type: "boolean"
  },
  suppressHydrationWarning: {
    required: !1,
    control: "boolean",
    type: "boolean"
  },
  tabIndex: {
    required: !1,
    control: "number",
    type: "number",
    description: "Overrides the browser's default tab order and follows the one specified instead."
  },
  title: {
    required: !1,
    control: "text",
    type: "string",
    description: "Text to be displayed in a tooltip when hovering over the element."
  },
  translate: {
    required: !1,
    control: "radio",
    type: "string",
    options: ["yes", "no"],
    description: "Specify whether an element's attribute values and the values of its text node children are to be translated when the page is localized, or whether to leave them unchanged."
  },
  typeof: { required: !1, control: "text", type: "string" },
  unselectable: {
    required: !1,
    control: "radio",
    type: "string",
    options: ["off", "on"]
  },
  value: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines a default value which will be displayed in the element on pageload."
  },
  vocab: { required: !1, control: "text", type: "string" },
  width: {
    required: !1,
    control: "number",
    type: "number",
    description: "Defines the image’s width in pixels."
  }
};
export {
  e as props
};
