// @webstudio-is/sdk-components-react/lib/__generated__/image.props.js
const e = {
  about: { required: !1, control: "text", type: "string" },
  accessKey: {
    required: !1,
    control: "text",
    type: "string",
    description: "Keyboard shortcut to activate or add focus to the element."
  },
  alt: {
    required: !1,
    control: "text",
    type: "string",
    description: "Text description of the image, which is very important for accessibility and search engine optimization. Screen readers read this description to users so they know what the image means. Alt text is also displayed on the page if the image can't be loaded for some reason."
  },
  autoCapitalize: {
    required: !1,
    control: "text",
    type: "string",
    description: "Sets whether input is automatically capitalized when entered by user."
  },
  autoCorrect: { required: !1, control: "text", type: "string" },
  autoFocus: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates that an element should be focused on page load, or when its parent dialog is displayed."
  },
  autoSave: { required: !1, control: "text", type: "string" },
  className: { required: !1, control: "text", type: "string" },
  color: {
    required: !1,
    control: "color",
    type: "string",
    description: "This attribute sets the text color using either a named color or a  color specified in the hexadecimal #RRGGBB format. Note: This is a legacy attribute. Please use the CSS color property instead."
  },
  content: {
    required: !1,
    control: "text",
    type: "string",
    description: "A value associated with http-equiv orname depending on the context."
  },
  contextMenu: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines the ID of a menu element which willserve as the element's context menu."
  },
  crossOrigin: {
    required: !1,
    control: "radio",
    type: "string",
    options: ["", "anonymous", "use-credentials"],
    description: "How the element handles cross-origin requests"
  },
  datatype: { required: !1, control: "text", type: "string" },
  decoding: {
    required: !1,
    control: "radio",
    type: "string",
    options: ["async", "auto", "sync"],
    description: "Indicates the preferred method to decode the image."
  },
  defaultChecked: { required: !1, control: "boolean", type: "boolean" },
  defaultValue: { required: !1, control: "text", type: "string" },
  dir: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines the text direction. Allowed values are ltr (Left-To-Right) or rtl (Right-To-Left)"
  },
  draggable: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Defines whether the element can be dragged."
  },
  fetchPriority: {
    required: !1,
    control: "radio",
    type: "string",
    options: ["high", "low", "auto"]
  },
  height: {
    required: !1,
    control: "number",
    type: "number",
    description: "Defines the image’s height in pixels."
  },
  hidden: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Prevents rendering of given element, while keeping child elements, e.g. script elements, active."
  },
  id: {
    required: !1,
    control: "text",
    type: "string",
    description: "Often used with CSS to style a specific element. The value of this attribute must be unique."
  },
  inputMode: {
    description: "Hints at the type of data that might be entered by the user while editing the element or its contents",
    required: !1,
    control: "select",
    type: "string",
    options: [
      "search",
      "text",
      "none",
      "tel",
      "url",
      "email",
      "numeric",
      "decimal"
    ]
  },
  is: {
    description: "Specify that a standard HTML element should behave like a defined custom built-in element",
    required: !1,
    control: "text",
    type: "string"
  },
  itemID: { required: !1, control: "text", type: "string" },
  itemProp: { required: !1, control: "text", type: "string" },
  itemRef: { required: !1, control: "text", type: "string" },
  itemScope: { required: !1, control: "boolean", type: "boolean" },
  itemType: { required: !1, control: "text", type: "string" },
  lang: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines the language used in the element."
  },
  loading: {
    required: !1,
    control: "radio",
    type: "string",
    defaultValue: "lazy",
    options: ["eager", "lazy"],
    description: "Determines whether the image will load as soon as possible (Eager), or when it scrolls into view (Lazy). Lazy loading is a great option for pages with many images because it can significantly reduce the time it takes for the page to load initially."
  },
  nonce: { required: !1, control: "text", type: "string" },
  optimize: {
    description: "Optimize the image for enhanced performance.",
    required: !1,
    control: "boolean",
    type: "boolean",
    defaultValue: !0
  },
  prefix: { required: !1, control: "text", type: "string" },
  property: { required: !1, control: "text", type: "string" },
  quality: { required: !1, control: "number", type: "number" },
  radioGroup: { required: !1, control: "text", type: "string" },
  referrerPolicy: {
    required: !1,
    control: "select",
    type: "string",
    options: [
      "",
      "origin",
      "no-referrer",
      "no-referrer-when-downgrade",
      "origin-when-cross-origin",
      "same-origin",
      "strict-origin",
      "strict-origin-when-cross-origin",
      "unsafe-url"
    ],
    description: "Specifies which referrer is sent when fetching the resource."
  },
  rel: {
    required: !1,
    control: "text",
    type: "string",
    description: "Specifies the relationship of the target object to the link object."
  },
  resource: { required: !1, control: "text", type: "string" },
  results: { required: !1, control: "number", type: "number" },
  rev: { required: !1, control: "text", type: "string" },
  role: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines an explicit role for an element for use by assistive technologies."
  },
  security: { required: !1, control: "text", type: "string" },
  sizes: { required: !1, control: "text", type: "string" },
  slot: {
    required: !1,
    control: "text",
    type: "string",
    description: "Assigns a slot in a shadow DOM shadow tree to an element."
  },
  spellCheck: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates whether spell checking is allowed for the element."
  },
  src: {
    required: !1,
    control: "text",
    type: "string",
    description: "The URL of the embeddable content."
  },
  srcSet: {
    required: !1,
    control: "text",
    type: "string",
    description: "One or more responsive image candidates."
  },
  suppressContentEditableWarning: {
    required: !1,
    control: "boolean",
    type: "boolean"
  },
  suppressHydrationWarning: {
    required: !1,
    control: "boolean",
    type: "boolean"
  },
  tabIndex: {
    required: !1,
    control: "number",
    type: "number",
    description: "Overrides the browser's default tab order and follows the one specified instead."
  },
  title: {
    required: !1,
    control: "text",
    type: "string",
    description: "Text to be displayed in a tooltip when hovering over the element."
  },
  translate: {
    required: !1,
    control: "radio",
    type: "string",
    options: ["yes", "no"],
    description: "Specify whether an element's attribute values and the values of its text node children are to be translated when the page is localized, or whether to leave them unchanged."
  },
  typeof: { required: !1, control: "text", type: "string" },
  unselectable: {
    required: !1,
    control: "radio",
    type: "string",
    options: ["on", "off"]
  },
  useMap: { required: !1, control: "text", type: "string" },
  vocab: { required: !1, control: "text", type: "string" },
  width: {
    required: !1,
    control: "number",
    type: "number",
    description: "Defines the image’s width in pixels."
  }
};
export {
  e as props
};
