// @webstudio-is/sdk-components-react/lib/__generated__/link.props.js
const e = {
  about: { required: !1, control: "text", type: "string" },
  accessKey: {
    required: !1,
    control: "text",
    type: "string",
    description: "Keyboard shortcut to activate or add focus to the element."
  },
  autoCapitalize: {
    required: !1,
    control: "text",
    type: "string",
    description: "Sets whether input is automatically capitalized when entered by user."
  },
  autoCorrect: { required: !1, control: "text", type: "string" },
  autoFocus: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates that an element should be focused on page load, or when its parent dialog is displayed."
  },
  autoSave: { required: !1, control: "text", type: "string" },
  className: { required: !1, control: "text", type: "string" },
  color: {
    required: !1,
    control: "color",
    type: "string",
    description: "This attribute sets the text color using either a named color or a  color specified in the hexadecimal #RRGGBB format. Note: This is a legacy attribute. Please use the CSS color property instead."
  },
  content: {
    required: !1,
    control: "text",
    type: "string",
    description: "A value associated with http-equiv orname depending on the context."
  },
  contextMenu: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines the ID of a menu element which willserve as the element's context menu."
  },
  datatype: { required: !1, control: "text", type: "string" },
  defaultChecked: { required: !1, control: "boolean", type: "boolean" },
  defaultValue: { required: !1, control: "text", type: "string" },
  dir: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines the text direction. Allowed values are ltr (Left-To-Right) or rtl (Right-To-Left)"
  },
  download: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates that the hyperlink is to be used for downloading a resource."
  },
  draggable: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Defines whether the element can be dragged."
  },
  hidden: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Prevents rendering of given element, while keeping child elements, e.g. script elements, active."
  },
  href: {
    required: !1,
    control: "text",
    type: "string",
    description: "The URL of a linked resource."
  },
  hrefLang: {
    required: !1,
    control: "text",
    type: "string",
    description: "Specifies the language of the linked resource."
  },
  id: {
    required: !1,
    control: "text",
    type: "string",
    description: "Often used with CSS to style a specific element. The value of this attribute must be unique."
  },
  inputMode: {
    description: "Hints at the type of data that might be entered by the user while editing the element or its contents",
    required: !1,
    control: "select",
    type: "string",
    options: [
      "search",
      "text",
      "none",
      "tel",
      "url",
      "email",
      "numeric",
      "decimal"
    ]
  },
  is: {
    description: "Specify that a standard HTML element should behave like a defined custom built-in element",
    required: !1,
    control: "text",
    type: "string"
  },
  itemID: { required: !1, control: "text", type: "string" },
  itemProp: { required: !1, control: "text", type: "string" },
  itemRef: { required: !1, control: "text", type: "string" },
  itemScope: { required: !1, control: "boolean", type: "boolean" },
  itemType: { required: !1, control: "text", type: "string" },
  lang: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines the language used in the element."
  },
  media: {
    required: !1,
    control: "text",
    type: "string",
    description: "Specifies a hint of the media for which the linked resource was designed."
  },
  nonce: { required: !1, control: "text", type: "string" },
  ping: {
    required: !1,
    control: "text",
    type: "string",
    description: "The ping attribute specifies a space-separated list of URLs to be notified if a user follows the hyperlink."
  },
  prefetch: {
    required: !1,
    control: "select",
    type: "string",
    options: ["none", "intent", "render", "viewport"],
    description: "Controls when and if the link prefetches the resources that the next page needs to make loading faster. “Intent” will prefetch when the link is hovered or focused. “Render” will prefetch when the link is rendered. “Viewport” will prefetch when the link is in the viewport. “None” will not prefetch."
  },
  prefix: { required: !1, control: "text", type: "string" },
  preventScrollReset: { required: !1, control: "boolean", type: "boolean" },
  property: { required: !1, control: "text", type: "string" },
  radioGroup: { required: !1, control: "text", type: "string" },
  referrerPolicy: {
    required: !1,
    control: "select",
    type: "string",
    options: [
      "",
      "no-referrer",
      "no-referrer-when-downgrade",
      "origin",
      "origin-when-cross-origin",
      "same-origin",
      "strict-origin",
      "strict-origin-when-cross-origin",
      "unsafe-url"
    ],
    description: "Specifies which referrer is sent when fetching the resource."
  },
  rel: {
    required: !1,
    control: "text",
    type: "string",
    description: "Specifies the relationship of the target object to the link object."
  },
  reloadDocument: { required: !1, control: "boolean", type: "boolean" },
  replace: { required: !1, control: "boolean", type: "boolean" },
  resource: { required: !1, control: "text", type: "string" },
  results: { required: !1, control: "number", type: "number" },
  rev: { required: !1, control: "text", type: "string" },
  role: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines an explicit role for an element for use by assistive technologies."
  },
  security: { required: !1, control: "text", type: "string" },
  slot: {
    required: !1,
    control: "text",
    type: "string",
    description: "Assigns a slot in a shadow DOM shadow tree to an element."
  },
  spellCheck: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates whether spell checking is allowed for the element."
  },
  suppressContentEditableWarning: {
    required: !1,
    control: "boolean",
    type: "boolean"
  },
  suppressHydrationWarning: {
    required: !1,
    control: "boolean",
    type: "boolean"
  },
  tabIndex: {
    required: !1,
    control: "number",
    type: "number",
    description: "Overrides the browser's default tab order and follows the one specified instead."
  },
  target: {
    required: !1,
    control: "select",
    type: "string",
    options: ["_self", "_blank", "_parent", "_top"],
    description: "Specifies where to open the linked document (in the case of an <a> element) or where to display the response received (in the case of a <form> element)"
  },
  title: {
    required: !1,
    control: "text",
    type: "string",
    description: "Text to be displayed in a tooltip when hovering over the element."
  },
  translate: {
    required: !1,
    control: "radio",
    type: "string",
    options: ["yes", "no"],
    description: "Specify whether an element's attribute values and the values of its text node children are to be translated when the page is localized, or whether to leave them unchanged."
  },
  type: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines the type of the element."
  },
  typeof: { required: !1, control: "text", type: "string" },
  unselectable: {
    required: !1,
    control: "radio",
    type: "string",
    options: ["on", "off"]
  },
  vocab: { required: !1, control: "text", type: "string" }
};
export {
  e as props
};
