// @webstudio-is/sdk-components-react/lib/__generated__/video.props.js
const e = {
  about: { required: !1, control: "text", type: "string" },
  accessKey: {
    required: !1,
    control: "text",
    type: "string",
    description: "Keyboard shortcut to activate or add focus to the element."
  },
  autoCapitalize: {
    required: !1,
    control: "text",
    type: "string",
    description: "Sets whether input is automatically capitalized when entered by user."
  },
  autoCorrect: { required: !1, control: "text", type: "string" },
  autoFocus: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates that an element should be focused on page load, or when its parent dialog is displayed."
  },
  autoPlay: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "The audio or video should play as soon as possible."
  },
  autoSave: { required: !1, control: "text", type: "string" },
  className: { required: !1, control: "text", type: "string" },
  color: {
    required: !1,
    control: "color",
    type: "string",
    description: "This attribute sets the text color using either a named color or a  color specified in the hexadecimal #RRGGBB format. Note: This is a legacy attribute. Please use the CSS color property instead."
  },
  content: {
    required: !1,
    control: "text",
    type: "string",
    description: "A value associated with http-equiv orname depending on the context."
  },
  contextMenu: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines the ID of a menu element which willserve as the element's context menu."
  },
  controls: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates whether the browser should show playback controls to the user."
  },
  controlsList: { required: !1, control: "text", type: "string" },
  crossOrigin: {
    required: !1,
    control: "radio",
    type: "string",
    options: ["", "anonymous", "use-credentials"],
    description: "How the element handles cross-origin requests"
  },
  datatype: { required: !1, control: "text", type: "string" },
  defaultChecked: { required: !1, control: "boolean", type: "boolean" },
  defaultValue: { required: !1, control: "text", type: "string" },
  dir: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines the text direction. Allowed values are ltr (Left-To-Right) or rtl (Right-To-Left)"
  },
  disablePictureInPicture: {
    required: !1,
    control: "boolean",
    type: "boolean"
  },
  disableRemotePlayback: {
    required: !1,
    control: "boolean",
    type: "boolean"
  },
  draggable: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Defines whether the element can be dragged."
  },
  height: {
    required: !1,
    control: "number",
    type: "number",
    description: "Defines the image’s height in pixels."
  },
  hidden: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Prevents rendering of given element, while keeping child elements, e.g. script elements, active."
  },
  id: {
    required: !1,
    control: "text",
    type: "string",
    description: "Often used with CSS to style a specific element. The value of this attribute must be unique."
  },
  inputMode: {
    description: "Hints at the type of data that might be entered by the user while editing the element or its contents",
    required: !1,
    control: "select",
    type: "string",
    options: [
      "search",
      "text",
      "none",
      "tel",
      "url",
      "email",
      "numeric",
      "decimal"
    ]
  },
  is: {
    description: "Specify that a standard HTML element should behave like a defined custom built-in element",
    required: !1,
    control: "text",
    type: "string"
  },
  itemID: { required: !1, control: "text", type: "string" },
  itemProp: { required: !1, control: "text", type: "string" },
  itemRef: { required: !1, control: "text", type: "string" },
  itemScope: { required: !1, control: "boolean", type: "boolean" },
  itemType: { required: !1, control: "text", type: "string" },
  lang: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines the language used in the element."
  },
  loop: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates whether the media should start playing from the start when it's finished."
  },
  mediaGroup: { required: !1, control: "text", type: "string" },
  muted: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates whether the audio will be initially silenced on page load."
  },
  nonce: { required: !1, control: "text", type: "string" },
  playsInline: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: `A Boolean attribute indicating that the video is to be played "inline"; that is, within the element's playback area. Note that the absence of this attribute does not imply that the video will always be played in fullscreen.`
  },
  poster: {
    required: !1,
    control: "text",
    type: "string",
    description: "A URL indicating a poster frame to show until the user plays or seeks."
  },
  prefix: { required: !1, control: "text", type: "string" },
  preload: {
    required: !1,
    control: "text",
    type: "string",
    description: "Indicates whether the whole resource, parts of it or nothing should be preloaded."
  },
  property: { required: !1, control: "text", type: "string" },
  radioGroup: { required: !1, control: "text", type: "string" },
  rel: {
    required: !1,
    control: "text",
    type: "string",
    description: "Specifies the relationship of the target object to the link object."
  },
  resource: { required: !1, control: "text", type: "string" },
  results: { required: !1, control: "number", type: "number" },
  rev: { required: !1, control: "text", type: "string" },
  role: {
    required: !1,
    control: "text",
    type: "string",
    description: "Defines an explicit role for an element for use by assistive technologies."
  },
  security: { required: !1, control: "text", type: "string" },
  slot: {
    required: !1,
    control: "text",
    type: "string",
    description: "Assigns a slot in a shadow DOM shadow tree to an element."
  },
  spellCheck: {
    required: !1,
    control: "boolean",
    type: "boolean",
    description: "Indicates whether spell checking is allowed for the element."
  },
  src: {
    required: !1,
    control: "text",
    type: "string",
    description: "The URL of the embeddable content."
  },
  suppressContentEditableWarning: {
    required: !1,
    control: "boolean",
    type: "boolean"
  },
  suppressHydrationWarning: {
    required: !1,
    control: "boolean",
    type: "boolean"
  },
  tabIndex: {
    required: !1,
    control: "number",
    type: "number",
    description: "Overrides the browser's default tab order and follows the one specified instead."
  },
  title: {
    required: !1,
    control: "text",
    type: "string",
    description: "Text to be displayed in a tooltip when hovering over the element."
  },
  translate: {
    required: !1,
    control: "radio",
    type: "string",
    options: ["yes", "no"],
    description: "Specify whether an element's attribute values and the values of its text node children are to be translated when the page is localized, or whether to leave them unchanged."
  },
  typeof: { required: !1, control: "text", type: "string" },
  unselectable: {
    required: !1,
    control: "radio",
    type: "string",
    options: ["on", "off"]
  },
  vocab: { required: !1, control: "text", type: "string" },
  width: {
    required: !1,
    control: "number",
    type: "number",
    description: "Defines the image’s width in pixels."
  }
};
export {
  e as props
};
