// @webstudio-is/sdk-components-react/lib/__generated__/vimeo.props.js
const e = {
  accessKey: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/accessKey)",
    required: !0,
    control: "text",
    type: "string"
  },
  accessKeyLabel: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/accessKeyLabel)",
    required: !0,
    control: "text",
    type: "string"
  },
  align: {
    description: "Sets or retrieves how the object is aligned with adjacent text.",
    required: !0,
    control: "text",
    type: "string"
  },
  allow: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLIFrameElement/allow)",
    required: !0,
    control: "text",
    type: "string"
  },
  allowFullscreen: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLIFrameElement/allowFullscreen)",
    required: !0,
    control: "boolean",
    type: "boolean"
  },
  autocapitalize: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/autocapitalize)",
    required: !0,
    control: "text",
    type: "string"
  },
  autofocus: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/autofocus)",
    required: !0,
    control: "boolean",
    type: "boolean"
  },
  baseURI: {
    description: `Returns node's node document's document base URL.

[MDN Reference](https://developer.mozilla.org/docs/Web/API/Node/baseURI)`,
    required: !0,
    control: "text",
    type: "string"
  },
  childElementCount: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/Document/childElementCount)",
    required: !0,
    control: "number",
    type: "number"
  },
  className: {
    description: `Returns the value of element's class content attribute. Can be set to change it.

[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/className)`,
    required: !0,
    control: "text",
    type: "string"
  },
  clientHeight: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/clientHeight)",
    required: !0,
    control: "number",
    type: "number"
  },
  clientLeft: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/clientLeft)",
    required: !0,
    control: "number",
    type: "number"
  },
  clientTop: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/clientTop)",
    required: !0,
    control: "number",
    type: "number"
  },
  clientWidth: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/clientWidth)",
    required: !0,
    control: "number",
    type: "number"
  },
  contentEditable: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/contentEditable)",
    required: !0,
    control: "text",
    type: "string"
  },
  currentCSSZoom: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/currentCSSZoom)",
    required: !0,
    control: "number",
    type: "number"
  },
  dir: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/dir)",
    required: !0,
    control: "text",
    type: "string"
  },
  draggable: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/draggable)",
    required: !0,
    control: "boolean",
    type: "boolean"
  },
  enterKeyHint: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/enterKeyHint)",
    required: !0,
    control: "text",
    type: "string"
  },
  frameBorder: {
    description: "Sets or retrieves whether to display a border for the frame.",
    required: !0,
    control: "text",
    type: "string"
  },
  height: {
    description: `Sets or retrieves the height of the object.

[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLIFrameElement/height)`,
    required: !0,
    control: "text",
    type: "string"
  },
  hidden: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/hidden)",
    required: !0,
    control: "boolean",
    type: "boolean"
  },
  id: {
    description: `Returns the value of element's id content attribute. Can be set to change it.

[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/id)`,
    required: !0,
    control: "text",
    type: "string"
  },
  inert: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/inert)",
    required: !0,
    control: "boolean",
    type: "boolean"
  },
  innerHTML: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/innerHTML)",
    required: !0,
    control: "text",
    type: "string"
  },
  innerText: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/innerText)",
    required: !0,
    control: "text",
    type: "string"
  },
  inputMode: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/inputMode)",
    required: !0,
    control: "text",
    type: "string"
  },
  isConnected: {
    description: `Returns true if node is connected and false otherwise.

[MDN Reference](https://developer.mozilla.org/docs/Web/API/Node/isConnected)`,
    required: !0,
    control: "boolean",
    type: "boolean"
  },
  isContentEditable: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/isContentEditable)",
    required: !0,
    control: "boolean",
    type: "boolean"
  },
  lang: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/lang)",
    required: !0,
    control: "text",
    type: "string"
  },
  loading: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLIFrameElement/loading)",
    required: !1,
    control: "text",
    type: "string",
    defaultValue: "lazy"
  },
  localName: {
    description: `Returns the local name.

[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/localName)`,
    required: !0,
    control: "text",
    type: "string"
  },
  longDesc: {
    description: "Sets or retrieves a URI to a long description of the object.",
    required: !0,
    control: "text",
    type: "string"
  },
  marginHeight: {
    description: "Sets or retrieves the top and bottom margin heights before displaying the text in a frame.",
    required: !0,
    control: "text",
    type: "string"
  },
  marginWidth: {
    description: "Sets or retrieves the left and right margin widths before displaying the text in a frame.",
    required: !0,
    control: "text",
    type: "string"
  },
  name: {
    description: `Sets or retrieves the frame name.

[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLIFrameElement/name)`,
    required: !0,
    control: "text",
    type: "string"
  },
  nodeName: {
    description: `Returns a string appropriate for the type of node.

[MDN Reference](https://developer.mozilla.org/docs/Web/API/Node/nodeName)`,
    required: !0,
    control: "text",
    type: "string"
  },
  nodeType: {
    description: `Returns the type of node.

[MDN Reference](https://developer.mozilla.org/docs/Web/API/Node/nodeType)`,
    required: !0,
    control: "number",
    type: "number"
  },
  nonce: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/nonce)",
    required: !1,
    control: "text",
    type: "string"
  },
  offsetHeight: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/offsetHeight)",
    required: !0,
    control: "number",
    type: "number"
  },
  offsetLeft: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/offsetLeft)",
    required: !0,
    control: "number",
    type: "number"
  },
  offsetTop: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/offsetTop)",
    required: !0,
    control: "number",
    type: "number"
  },
  offsetWidth: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/offsetWidth)",
    required: !0,
    control: "number",
    type: "number"
  },
  outerHTML: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/outerHTML)",
    required: !0,
    control: "text",
    type: "string"
  },
  outerText: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/outerText)",
    required: !0,
    control: "text",
    type: "string"
  },
  referrerPolicy: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLIFrameElement/referrerPolicy)",
    required: !0,
    control: "select",
    type: "string",
    options: [
      "",
      "no-referrer",
      "no-referrer-when-downgrade",
      "origin",
      "origin-when-cross-origin",
      "same-origin",
      "strict-origin",
      "strict-origin-when-cross-origin",
      "unsafe-url"
    ]
  },
  role: {
    required: !0,
    control: "text",
    type: "string",
    description: "Defines an explicit role for an element for use by assistive technologies."
  },
  scrollHeight: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/scrollHeight)",
    required: !0,
    control: "number",
    type: "number"
  },
  scrolling: {
    description: "Sets or retrieves whether the frame can be scrolled.",
    required: !0,
    control: "text",
    type: "string"
  },
  scrollLeft: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/scrollLeft)",
    required: !0,
    control: "number",
    type: "number"
  },
  scrollTop: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/scrollTop)",
    required: !0,
    control: "number",
    type: "number"
  },
  scrollWidth: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/scrollWidth)",
    required: !0,
    control: "number",
    type: "number"
  },
  slot: {
    description: `Returns the value of element's slot content attribute. Can be set to change it.

[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/slot)`,
    required: !0,
    control: "text",
    type: "string"
  },
  spellcheck: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/spellcheck)",
    required: !0,
    control: "boolean",
    type: "boolean"
  },
  src: {
    description: `Sets or retrieves a URL to be loaded by the object.

[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLIFrameElement/src)`,
    required: !0,
    control: "text",
    type: "string"
  },
  srcdoc: {
    description: `Sets or retrives the content of the page that is to contain.

[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLIFrameElement/srcdoc)`,
    required: !0,
    control: "text",
    type: "string"
  },
  tabIndex: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/tabIndex)",
    required: !0,
    control: "number",
    type: "number"
  },
  tagName: {
    description: `Returns the HTML-uppercased qualified name.

[MDN Reference](https://developer.mozilla.org/docs/Web/API/Element/tagName)`,
    required: !0,
    control: "text",
    type: "string"
  },
  title: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/title)",
    required: !0,
    control: "text",
    type: "string"
  },
  translate: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/translate)",
    required: !0,
    control: "boolean",
    type: "boolean"
  },
  width: {
    description: `Sets or retrieves the width of the object.

[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLIFrameElement/width)`,
    required: !0,
    control: "text",
    type: "string"
  },
  writingSuggestions: {
    description: "[MDN Reference](https://developer.mozilla.org/docs/Web/API/HTMLElement/writingSuggestions)",
    required: !0,
    control: "text",
    type: "string"
  }
};
export {
  e as props
};
