// @webstudio-is/sdk-components-animation/lib/__generated__/animate-text.props.js
const e = {
  className: {
    required: !1,
    control: "text",
    type: "string",
    description: "Classes to which the element belongs"
  },
  easing: {
    description: "Easing function applied within the sliding window.",
    required: !1,
    control: "select",
    type: "string",
    defaultValue: "linear",
    options: [
      "linear",
      "easeIn",
      "easeInCubic",
      "easeInQuart",
      "easeOut",
      "easeOutCubic",
      "easeOutQuart",
      "ease",
      "easeInOutCubic",
      "easeInOutQuart"
    ]
  },
  slidingWindow: {
    description: `Size of the sliding window for the animation:
- 0: Typewriter effect (no animation).
- (0..1]: Animates one part of the text at a time.
- (1..n]: Animates multiple parts of the text within the sliding window.`,
    required: !1,
    control: "number",
    type: "number",
    defaultValue: 5
  },
  splitBy: {
    description: "Defines how the text is split for animation (e.g., by character, space, or symbol).",
    required: !1,
    control: "select",
    type: "string",
    defaultValue: "char",
    options: ["char", "space", 'symbol "#"', 'symbol "~"']
  }
};
export {
  e as props
};
