// @webstudio-is/sdk/lib/__generated__/normalize.css.js
// src/__generated__/normalize.css.ts
var div = [
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  { property: "outline-width", value: { type: "unit", unit: "px", value: 1 } }
];
var address = div;
var article = div;
var aside = div;
var figure = div;
var footer = div;
var header = div;
var main = div;
var nav = div;
var section = div;
var form = div;
var label = div;
var time = div;
var h1 = div;
var h2 = div;
var h3 = div;
var h4 = div;
var h5 = div;
var h6 = div;
var i = div;
var img = div;
var a = div;
var li = div;
var ul = div;
var ol = div;
var p = div;
var span = div;
var html = [
  { property: "display", value: { type: "keyword", value: "grid" } },
  { property: "min-height", value: { type: "unit", unit: "%", value: 100 } },
  {
    property: "font-family",
    value: { type: "fontFamily", value: ["Arial", "Roboto", "sans-serif"] }
  },
  { property: "font-size", value: { type: "unit", unit: "px", value: 16 } },
  {
    property: "line-height",
    value: { type: "unit", unit: "number", value: 1.2 }
  },
  {
    property: "white-space-collapse",
    value: { type: "keyword", value: "preserve" }
  }
];
var body = [
  { property: "margin-top", value: { type: "unit", unit: "number", value: 0 } },
  {
    property: "margin-right",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "margin-bottom",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "margin-left",
    value: { type: "unit", unit: "number", value: 0 }
  },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "-webkit-font-smoothing",
    value: { type: "keyword", value: "antialiased" }
  },
  {
    property: "-moz-osx-font-smoothing",
    value: { type: "keyword", value: "grayscale" }
  }
];
var hr = [
  { property: "height", value: { type: "unit", unit: "number", value: 0 } },
  { property: "color", value: { type: "keyword", value: "inherit" } },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } }
];
var b = [
  {
    property: "font-weight",
    value: { type: "unit", unit: "number", value: 700 }
  },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
var strong = b;
var code = [
  {
    property: "font-family",
    value: {
      type: "fontFamily",
      value: [
        "ui-monospace",
        "SFMono-Regular",
        "Consolas",
        "Liberation Mono",
        "Menlo",
        "monospace"
      ]
    }
  },
  { property: "font-size", value: { type: "unit", unit: "em", value: 1 } },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
var kbd = code;
var samp = code;
var pre = code;
var small = [
  { property: "font-size", value: { type: "unit", unit: "%", value: 80 } },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
var sub = [
  { property: "font-size", value: { type: "unit", unit: "%", value: 75 } },
  {
    property: "line-height",
    value: { type: "unit", unit: "number", value: 0 }
  },
  { property: "position", value: { type: "keyword", value: "relative" } },
  { property: "vertical-align", value: { type: "keyword", value: "baseline" } },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  { property: "bottom", value: { type: "unit", unit: "em", value: -0.25 } }
];
var sup = [
  { property: "font-size", value: { type: "unit", unit: "%", value: 75 } },
  {
    property: "line-height",
    value: { type: "unit", unit: "number", value: 0 }
  },
  { property: "position", value: { type: "keyword", value: "relative" } },
  { property: "vertical-align", value: { type: "keyword", value: "baseline" } },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  { property: "top", value: { type: "unit", unit: "em", value: -0.5 } }
];
var table = [
  {
    property: "text-indent",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-top-color",
    value: { type: "keyword", value: "inherit" }
  },
  {
    property: "border-right-color",
    value: { type: "keyword", value: "inherit" }
  },
  {
    property: "border-bottom-color",
    value: { type: "keyword", value: "inherit" }
  },
  {
    property: "border-left-color",
    value: { type: "keyword", value: "inherit" }
  },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } }
];
var input = [
  { property: "font-family", value: { type: "keyword", value: "inherit" } },
  { property: "font-size", value: { type: "unit", unit: "%", value: 100 } },
  {
    property: "line-height",
    value: { type: "unit", unit: "number", value: 1.15 }
  },
  { property: "margin-top", value: { type: "unit", unit: "number", value: 0 } },
  {
    property: "margin-right",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "margin-bottom",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "margin-left",
    value: { type: "unit", unit: "number", value: 0 }
  },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  { property: "border-top-style", value: { type: "keyword", value: "solid" } },
  {
    property: "border-right-style",
    value: { type: "keyword", value: "solid" }
  },
  {
    property: "border-bottom-style",
    value: { type: "keyword", value: "solid" }
  },
  { property: "border-left-style", value: { type: "keyword", value: "solid" } }
];
var textarea = input;
var optgroup = [
  { property: "font-family", value: { type: "keyword", value: "inherit" } },
  { property: "font-size", value: { type: "unit", unit: "%", value: 100 } },
  {
    property: "line-height",
    value: { type: "unit", unit: "number", value: 1.15 }
  },
  { property: "margin-top", value: { type: "unit", unit: "number", value: 0 } },
  {
    property: "margin-right",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "margin-bottom",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "margin-left",
    value: { type: "unit", unit: "number", value: 0 }
  },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
var radio = [
  { property: "font-family", value: { type: "keyword", value: "inherit" } },
  { property: "font-size", value: { type: "unit", unit: "%", value: 100 } },
  {
    property: "line-height",
    value: { type: "unit", unit: "number", value: 1.15 }
  },
  { property: "margin-top", value: { type: "unit", unit: "number", value: 0 } },
  {
    property: "margin-right",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "margin-bottom",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "margin-left",
    value: { type: "unit", unit: "number", value: 0 }
  },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  { property: "border-top-style", value: { type: "keyword", value: "none" } },
  { property: "border-right-style", value: { type: "keyword", value: "none" } },
  {
    property: "border-bottom-style",
    value: { type: "keyword", value: "none" }
  },
  { property: "border-left-style", value: { type: "keyword", value: "none" } }
];
var checkbox = radio;
var button = [
  { property: "font-family", value: { type: "keyword", value: "inherit" } },
  { property: "font-size", value: { type: "unit", unit: "%", value: 100 } },
  {
    property: "line-height",
    value: { type: "unit", unit: "number", value: 1.15 }
  },
  { property: "margin-top", value: { type: "unit", unit: "number", value: 0 } },
  {
    property: "margin-right",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "margin-bottom",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "margin-left",
    value: { type: "unit", unit: "number", value: 0 }
  },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  { property: "border-top-style", value: { type: "keyword", value: "solid" } },
  {
    property: "border-right-style",
    value: { type: "keyword", value: "solid" }
  },
  {
    property: "border-bottom-style",
    value: { type: "keyword", value: "solid" }
  },
  { property: "border-left-style", value: { type: "keyword", value: "solid" } },
  { property: "text-transform", value: { type: "keyword", value: "none" } }
];
var select = button;
var legend = [
  {
    property: "padding-top",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "padding-right",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "padding-bottom",
    value: { type: "unit", unit: "number", value: 0 }
  },
  {
    property: "padding-left",
    value: { type: "unit", unit: "number", value: 0 }
  },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
var progress = [
  { property: "vertical-align", value: { type: "keyword", value: "baseline" } },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
var summary = [
  { property: "display", value: { type: "keyword", value: "list-item" } },
  { property: "box-sizing", value: { type: "keyword", value: "border-box" } },
  {
    property: "border-top-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-right-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-bottom-width",
    value: { type: "unit", unit: "px", value: 1 }
  },
  {
    property: "border-left-width",
    value: { type: "unit", unit: "px", value: 1 }
  }
];
export {
  a,
  address,
  article,
  aside,
  b,
  body,
  button,
  checkbox,
  code,
  div,
  figure,
  footer,
  form,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  header,
  hr,
  html,
  i,
  img,
  input,
  kbd,
  label,
  legend,
  li,
  main,
  nav,
  ol,
  optgroup,
  p,
  pre,
  progress,
  radio,
  samp,
  section,
  select,
  small,
  span,
  strong,
  sub,
  summary,
  sup,
  table,
  textarea,
  time,
  ul
};
