// @webstudio-is/sdk-components-react/lib/blockquote.ws.js
import { props as t } from "./__generated__/blockquote.props.js";
const e = {
  blockquote: [
    {
      property: "margin-top",
      value: { type: "unit", value: 0, unit: "number" }
    },
    {
      property: "margin-right",
      value: { type: "unit", value: 0, unit: "number" }
    },
    {
      property: "margin-bottom",
      value: { type: "unit", value: 10, unit: "px" }
    },
    {
      property: "margin-left",
      value: { type: "unit", value: 0, unit: "number" }
    },
    {
      property: "padding-top",
      value: { type: "unit", value: 10, unit: "px" }
    },
    {
      property: "padding-bottom",
      value: { type: "unit", value: 10, unit: "px" }
    },
    {
      property: "padding-left",
      value: { type: "unit", value: 20, unit: "px" }
    },
    {
      property: "padding-right",
      value: { type: "unit", value: 20, unit: "px" }
    },
    {
      property: "border-left-width",
      value: { type: "unit", value: 5, unit: "px" }
    },
    {
      property: "border-left-style",
      value: { type: "keyword", value: "solid" }
    },
    {
      property: "border-left-color",
      value: { type: "rgb", r: 226, g: 226, b: 226, alpha: 1 }
    }
  ]
}, r = {
  presetStyle: e,
  initialProps: ["id", "class", "cite"],
  props: t
};
export {
  r as meta
};
