// @webstudio-is/sdk-components-react/lib/blockquote.ws.js
import { BlockquoteIcon as t } from "@webstudio-is/icons/svg";
import { defaultStates as e } from "@webstudio-is/sdk";
import { props as p } from "./__generated__/blockquote.props.js";
const r = {
  blockquote: [
    {
      property: "margin-top",
      value: { type: "unit", value: 0, unit: "number" }
    },
    {
      property: "margin-right",
      value: { type: "unit", value: 0, unit: "number" }
    },
    {
      property: "margin-bottom",
      value: { type: "unit", value: 10, unit: "px" }
    },
    {
      property: "margin-left",
      value: { type: "unit", value: 0, unit: "number" }
    },
    {
      property: "padding-top",
      value: { type: "unit", value: 10, unit: "px" }
    },
    {
      property: "padding-bottom",
      value: { type: "unit", value: 10, unit: "px" }
    },
    {
      property: "padding-left",
      value: { type: "unit", value: 20, unit: "px" }
    },
    {
      property: "padding-right",
      value: { type: "unit", value: 20, unit: "px" }
    },
    {
      property: "border-left-width",
      value: { type: "unit", value: 5, unit: "px" }
    },
    {
      property: "border-left-style",
      value: { type: "keyword", value: "solid" }
    },
    {
      property: "border-left-color",
      value: { type: "rgb", r: 226, g: 226, b: 226, alpha: 1 }
    }
  ]
}, i = {
  placeholder: "Blockquote",
  icon: t,
  states: e,
  presetStyle: r
}, l = {
  props: p,
  initialProps: ["id", "className", "cite"]
};
export {
  i as meta,
  l as propsMeta
};
