// @webstudio-is/sdk-components-react/lib/box.ws.js
import { defaultStates as e } from "@webstudio-is/sdk";
import { section as t, nav as r, main as a, header as i, footer as o, figure as s, aside as n, article as d, address as p, div as c } from "@webstudio-is/sdk/normalize.css";
import { props as g } from "./__generated__/box.props.js";
const u = {
  category: "general",
  description: "A container for content. By default this is a Div, but the tag can be changed in settings.",
  states: e,
  presetStyle: {
    div: c,
    address: p,
    article: d,
    aside: n,
    figure: s,
    footer: o,
    header: i,
    main: a,
    nav: r,
    section: t
  },
  order: 0,
  initialProps: ["tag", "id", "class"],
  props: {
    ...g,
    tag: {
      required: !0,
      control: "tag",
      type: "string",
      options: [
        "div",
        "header",
        "footer",
        "nav",
        "main",
        "section",
        "article",
        "aside",
        "address",
        "figure",
        "span"
      ]
    }
  }
};
export {
  u as meta
};
