// @webstudio-is/sdk-components-react/lib/box.ws.js
import { BoxIcon as t } from "@webstudio-is/icons/svg";
import { defaultStates as e } from "@webstudio-is/sdk";
import { div as r, address as o, article as a, aside as i, figure as s, footer as n, header as d, main as p, nav as c, section as m } from "@webstudio-is/sdk/normalize.css";
import { props as f } from "./__generated__/box.props.js";
const v = {
  category: "general",
  description: "A container for content. By default this is a Div, but the tag can be changed in settings.",
  icon: t,
  states: e,
  presetStyle: {
    div: r,
    address: o,
    article: a,
    aside: i,
    figure: s,
    footer: n,
    header: d,
    main: p,
    nav: c,
    section: m
  },
  order: 0
}, y = {
  props: {
    ...f,
    tag: {
      required: !0,
      control: "tag",
      type: "string",
      options: [
        "div",
        "header",
        "footer",
        "nav",
        "main",
        "section",
        "article",
        "aside",
        "address",
        "figure",
        "span"
      ]
    }
  },
  initialProps: ["tag", "id", "className"]
};
export {
  v as meta,
  y as propsMeta
};
