// @webstudio-is/sdk-components-react/lib/code-text.ws.js
import { BracesIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as t } from "@webstudio-is/sdk";
import { code as o } from "@webstudio-is/sdk/normalize.css";
import { props as r } from "./__generated__/code-text.props.js";
const p = {
  code: [
    ...o,
    {
      property: "display",
      value: { type: "keyword", value: "block" }
    },
    {
      property: "white-space-collapse",
      value: { type: "keyword", value: "preserve" }
    },
    {
      property: "text-wrap-mode",
      value: { type: "keyword", value: "wrap" }
    },
    {
      property: "padding-left",
      value: { type: "unit", value: 0.2, unit: "em" }
    },
    {
      property: "padding-right",
      value: { type: "unit", value: 0.2, unit: "em" }
    },
    {
      property: "background-color",
      value: { type: "rgb", r: 238, g: 238, b: 238, alpha: 1 }
    }
  ]
}, c = {
  category: "general",
  description: "Use this component when you want to display code as text on the page.",
  icon: e,
  contentModel: {
    category: "instance",
    children: []
  },
  states: t,
  presetStyle: p,
  order: 5,
  initialProps: ["id", "class", "lang", "code"],
  props: {
    ...r,
    code: {
      required: !0,
      control: "codetext",
      type: "string"
    }
  }
};
export {
  c as meta
};
