// @webstudio-is/sdk-components-react/lib/input.ws.js
import { defaultStates as e } from "@webstudio-is/sdk";
import { input as l } from "@webstudio-is/sdk/normalize.css";
import { props as t } from "./__generated__/input.props.js";
const r = {
  input: [
    ...l,
    {
      property: "display",
      value: { type: "keyword", value: "block" }
    }
  ]
}, s = {
  category: "forms",
  label: "Text Input",
  description: "A single-line text input for collecting string data from your users.",
  presetStyle: r,
  order: 3,
  states: [
    ...e,
    { selector: "::placeholder", label: "Placeholder" },
    { selector: ":valid", label: "Valid" },
    { selector: ":invalid", label: "Invalid" },
    { selector: ":required", label: "Required" },
    { selector: ":optional", label: "Optional" }
    // Additional states will go into submenu
    //{ selector: ":disabled", label: "Disabled" },
    //{ selector: ":enabled", label: "Enabled" },
    //{ selector: ":read-only", label: "Read Only" },
    //{ selector: ":read-write", label: "Read Write" },
  ],
  initialProps: [
    "id",
    "class",
    "name",
    "value",
    "type",
    "placeholder",
    "required",
    "autofocus"
  ],
  props: t
};
export {
  s as meta
};
