// @webstudio-is/sdk-components-react/lib/list.ws.js
import { ListIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as t } from "@webstudio-is/sdk";
import { ol as r, ul as o } from "@webstudio-is/sdk/normalize.css";
import { props as p } from "./__generated__/list.props.js";
const a = {
  ol: [
    ...r,
    {
      property: "margin-top",
      value: { type: "keyword", value: "0" }
    },
    {
      property: "margin-bottom",
      value: { type: "keyword", value: "10px" }
    },
    {
      property: "padding-left",
      value: { type: "keyword", value: "40px" }
    }
  ],
  ul: [
    ...o,
    {
      property: "margin-top",
      value: { type: "keyword", value: "0" }
    },
    {
      property: "margin-bottom",
      value: { type: "keyword", value: "10px" }
    },
    {
      property: "padding-left",
      value: { type: "keyword", value: "40px" }
    }
  ]
}, d = {
  icon: e,
  states: t,
  presetStyle: a
}, s = {
  props: p,
  initialProps: ["id", "className", "ordered", "start", "reversed"]
};
export {
  d as meta,
  s as propsMeta
};
