// @webstudio-is/sdk-components-react/lib/props.js
import { propsMeta as p } from "./slot.ws.js";
import { propsMeta as t } from "./fragment.ws.js";
import { propsMeta as s } from "./html-embed.ws.js";
import { propsMeta as x } from "./markdown-embed.ws.js";
import { propsMeta as f } from "./body.ws.js";
import { propsMeta as d } from "./box.ws.js";
import { propsMeta as l } from "./text.ws.js";
import { propsMeta as S } from "./heading.ws.js";
import { propsMeta as c } from "./paragraph.ws.js";
import { propsMeta as b } from "./link.ws.js";
import { propsMeta as H } from "./rich-text-link.ws.js";
import { propsMeta as g } from "./span.ws.js";
import { propsMeta as V } from "./bold.ws.js";
import { propsMeta as F } from "./italic.ws.js";
import { propsMeta as R } from "./superscript.ws.js";
import { propsMeta as y } from "./subscript.ws.js";
import { propsMeta as E } from "./button.ws.js";
import { propsMeta as q } from "./input.ws.js";
import { propsMeta as N } from "./webhook-form.ws.js";
import { propsMeta as Y } from "./form.ws.js";
import { propsMeta as z } from "./image.ws.js";
import { propsMeta as D } from "./blockquote.ws.js";
import { propsMeta as J } from "./list.ws.js";
import { propsMeta as Q } from "./list-item.ws.js";
import { propsMeta as W } from "./separator.ws.js";
import { propsMeta as _ } from "./code-text.ws.js";
import { propsMeta as oo } from "./label.ws.js";
import { propsMeta as po } from "./textarea.ws.js";
import { propsMeta as to } from "./radio-button.ws.js";
import { propsMeta as so } from "./checkbox.ws.js";
import { propsMeta as xo } from "./vimeo.ws.js";
import { propsMeta as fo } from "./youtube.ws.js";
import { propsMeta as no } from "./vimeo-preview-image.ws.js";
import { propsMeta as uo } from "./vimeo-play-button.ws.js";
import { propsMeta as To } from "./vimeo-spinner.ws.js";
import { propsMeta as Bo } from "./xml-node.ws.js";
import { propsMeta as ko } from "./xml-time.ws.js";
import { propsMeta as Lo } from "./time.ws.js";
import { propsMeta as Io } from "./select.ws.js";
import { propsMeta as ho } from "./option.ws.js";
import { propsMeta as Po } from "./head-slot.ws.js";
import { propsMeta as wo } from "./head-link.ws.js";
import { propsMeta as Co } from "./head-meta.ws.js";
import { propsMeta as Xo } from "./head-title.ws.js";
import { propsMeta as vo } from "./video.ws.js";
export {
  D as Blockquote,
  f as Body,
  V as Bold,
  d as Box,
  E as Button,
  so as Checkbox,
  _ as CodeText,
  N as Form,
  t as Fragment,
  wo as HeadLink,
  Co as HeadMeta,
  Po as HeadSlot,
  Xo as HeadTitle,
  S as Heading,
  s as HtmlEmbed,
  z as Image,
  q as Input,
  F as Italic,
  oo as Label,
  b as Link,
  J as List,
  Q as ListItem,
  x as MarkdownEmbed,
  ho as Option,
  c as Paragraph,
  to as RadioButton,
  Y as RemixForm,
  H as RichTextLink,
  Io as Select,
  W as Separator,
  p as Slot,
  g as Span,
  y as Subscript,
  R as Superscript,
  l as Text,
  po as Textarea,
  Lo as Time,
  vo as Video,
  xo as Vimeo,
  uo as VimeoPlayButton,
  no as VimeoPreviewImage,
  To as VimeoSpinner,
  Bo as XmlNode,
  ko as XmlTime,
  fo as YouTube
};
