// @webstudio-is/sdk-components-react/lib/radio-button.ws.js
import { RadioCheckedIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as t } from "@webstudio-is/sdk";
import { radio as o } from "@webstudio-is/sdk/normalize.css";
import { props as r } from "./__generated__/radio-button.props.js";
const a = {
  input: [
    ...o,
    {
      property: "margin-right",
      value: { type: "unit", unit: "em", value: 0.5 }
    }
  ]
}, s = {
  label: "Radio",
  icon: e,
  presetStyle: a,
  states: [
    ...t,
    { selector: ":checked", label: "Checked" },
    { selector: ":required", label: "Required" },
    { selector: ":optional", label: "Optional" }
    // Additional states will go into submenu
    //{ selector: ":disabled", label: "Disabled" },
    //{ selector: ":enabled", label: "Enabled" },
    //{ selector: ":read-only", label: "Read Only" },
    //{ selector: ":read-write", label: "Read Write" },
  ]
}, m = {
  props: r,
  initialProps: ["id", "className", "name", "value", "required", "checked"]
};
export {
  s as meta,
  m as propsMeta
};
