// @webstudio-is/sdk-components-react/lib/separator.ws.js
import { MinusIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as o } from "@webstudio-is/sdk";
import { hr as r } from "@webstudio-is/sdk/normalize.css";
import { props as t } from "./__generated__/separator.props.js";
const p = {
  hr: [
    ...r,
    {
      property: "height",
      value: { type: "keyword", value: "1px" }
    },
    {
      property: "background-color",
      value: { type: "keyword", value: "gray" }
    },
    {
      property: "border-top-style",
      value: { type: "keyword", value: "none" }
    },
    {
      property: "border-right-style",
      value: { type: "keyword", value: "none" }
    },
    {
      property: "border-left-style",
      value: { type: "keyword", value: "none" }
    },
    {
      property: "border-bottom-style",
      value: { type: "keyword", value: "none" }
    }
  ]
}, n = {
  category: "general",
  description: "Used to visually divide sections of content, helping to improve readability and organization within a webpage.",
  icon: e,
  states: o,
  presetStyle: p,
  order: 3
}, s = {
  props: t,
  initialProps: ["id", "className"]
};
export {
  n as meta,
  s as propsMeta
};
