// @webstudio-is/sdk-components-react/lib/separator.ws.js
import { defaultStates as e } from "@webstudio-is/sdk";
import { hr as r } from "@webstudio-is/sdk/normalize.css";
import { props as o } from "./__generated__/separator.props.js";
const t = {
  hr: [
    ...r,
    {
      property: "height",
      value: { type: "keyword", value: "1px" }
    },
    {
      property: "background-color",
      value: { type: "keyword", value: "gray" }
    },
    {
      property: "border-top-style",
      value: { type: "keyword", value: "none" }
    },
    {
      property: "border-right-style",
      value: { type: "keyword", value: "none" }
    },
    {
      property: "border-left-style",
      value: { type: "keyword", value: "none" }
    },
    {
      property: "border-bottom-style",
      value: { type: "keyword", value: "none" }
    }
  ]
}, y = {
  category: "general",
  description: "Used to visually divide sections of content, helping to improve readability and organization within a webpage.",
  states: e,
  presetStyle: t,
  order: 3,
  initialProps: ["id", "class"],
  props: o
};
export {
  y as meta
};
