// @webstudio-is/sdk-components-react/lib/templates.js
import { meta as o } from "./content-embed.template.js";
import { meta as m } from "./markdown-embed.template.js";
import { meta as s } from "./text.template.js";
import { meta as p } from "./heading.template.js";
import { meta as n } from "./paragraph.template.js";
import { meta as i } from "./link.template.js";
import { meta as u } from "./button.template.js";
import { meta as l } from "./webhook-form.template.js";
import { meta as c } from "./blockquote.template.js";
import { meta as g } from "./list.template.js";
import { meta as C } from "./list-item.template.js";
import { meta as H } from "./label.template.js";
import { meta as T } from "./radio-button.template.js";
import { meta as w } from "./checkbox.template.js";
import { meta as I } from "./vimeo.template.js";
import { meta as P } from "./youtube.template.js";
import { meta as V } from "./select.template.js";
import { meta as j } from "./head-slot.template.js";
export {
  c as Blockquote,
  u as Button,
  w as Checkbox,
  o as ContentEmbed,
  l as Form,
  j as HeadSlot,
  p as Heading,
  H as Label,
  i as Link,
  g as List,
  C as ListItem,
  m as MarkdownEmbed,
  n as Paragraph,
  T as RadioButton,
  V as Select,
  s as Text,
  I as Vimeo,
  P as YouTube
};
