// @webstudio-is/sdk-components-react/lib/textarea.ws.js
import { defaultStates as e } from "@webstudio-is/sdk";
import { textarea as l } from "@webstudio-is/sdk/normalize.css";
import { props as r } from "./__generated__/textarea.props.js";
const t = {
  textarea: [
    ...l,
    // resize doesn't work well while on canvas
    { property: "resize", value: { type: "keyword", value: "none" } },
    {
      property: "display",
      value: { type: "keyword", value: "block" }
    }
  ]
}, s = {
  category: "forms",
  label: "Text Area",
  description: "A multi-line text input for collecting longer string data from your users.",
  presetStyle: t,
  order: 4,
  contentModel: {
    category: "instance",
    children: []
  },
  states: [
    ...e,
    { selector: "::placeholder", label: "Placeholder" },
    { selector: ":valid", label: "Valid" },
    { selector: ":invalid", label: "Invalid" },
    { selector: ":required", label: "Required" },
    { selector: ":optional", label: "Optional" }
    // Additional states will go into submenu
    //{ selector: ":disabled", label: "Disabled" },
    //{ selector: ":enabled", label: "Enabled" },
    //{ selector: ":read-only", label: "Read Only" },
    //{ selector: ":read-write", label: "Read Write" },
  ],
  initialProps: [
    "id",
    "class",
    "name",
    "value",
    "placeholder",
    "required",
    "autofocus"
  ],
  props: r
};
export {
  s as meta
};
