// @webstudio-is/sdk-components-react/lib/textarea.ws.js
import { FormTextAreaIcon as e } from "@webstudio-is/icons/svg";
import { defaultStates as r } from "@webstudio-is/sdk";
import { textarea as o } from "@webstudio-is/sdk/normalize.css";
import { props as t } from "./__generated__/textarea.props.js";
const l = {
  textarea: [
    ...o,
    // resize doesn't work well while on canvas
    { property: "resize", value: { type: "keyword", value: "none" } },
    {
      property: "display",
      value: { type: "keyword", value: "block" }
    }
  ]
}, n = {
  category: "forms",
  label: "Text Area",
  description: "A multi-line text input for collecting longer string data from your users.",
  icon: e,
  presetStyle: l,
  order: 4,
  contentModel: {
    category: "instance",
    children: []
  },
  states: [
    ...r,
    { selector: "::placeholder", label: "Placeholder" },
    { selector: ":valid", label: "Valid" },
    { selector: ":invalid", label: "Invalid" },
    { selector: ":required", label: "Required" },
    { selector: ":optional", label: "Optional" }
    // Additional states will go into submenu
    //{ selector: ":disabled", label: "Disabled" },
    //{ selector: ":enabled", label: "Enabled" },
    //{ selector: ":read-only", label: "Read Only" },
    //{ selector: ":read-write", label: "Read Write" },
  ]
}, p = {
  props: t,
  initialProps: [
    "id",
    "className",
    "name",
    "value",
    "placeholder",
    "required",
    "autoFocus"
  ]
};
export {
  n as meta,
  p as propsMeta
};
