// @webstudio-is/sdk-components-react/lib/webhook-form.template.js
import { jsxs as r, jsx as t } from "react/jsx-runtime";
import { $ as e, ActionValue as s, PlaceholderValue as o, Variable as i, expression as a } from "@webstudio-is/template";
const n = new i("formState", "initial"), w = {
  category: "data",
  order: 1,
  description: "Collect user data and send it to any webhook.",
  template: /* @__PURE__ */ r(
    e.Form,
    {
      state: a`${n}`,
      onStateChange: new s(["state"], a`${n} = state`),
      children: [
        /* @__PURE__ */ r(
          e.Box,
          {
            "ws:label": "Form Content",
            "ws:show": a`${n} === 'initial' || ${n} === 'error'`,
            children: [
              /* @__PURE__ */ t(e.Label, { children: new o("Name") }),
              /* @__PURE__ */ t(e.Input, { name: "name" }),
              /* @__PURE__ */ t(e.Label, { children: new o("Email") }),
              /* @__PURE__ */ t(e.Input, { name: "email" }),
              /* @__PURE__ */ t(e.Button, { children: new o("Submit") })
            ]
          }
        ),
        /* @__PURE__ */ t(
          e.Box,
          {
            "ws:label": "Success Message",
            "ws:show": a`${n} === 'success'`,
            children: new o("Thank you for getting in touch!")
          }
        ),
        /* @__PURE__ */ t(
          e.Box,
          {
            "ws:label": "Error Message",
            "ws:show": a`${n} === 'error'`,
            children: new o("Sorry, something went wrong.")
          }
        )
      ]
    }
  )
};
export {
  w as meta
};
