// @webstudio-is/sdk-components-react/lib/webhook-form.template.js
import { jsxs as r, jsx as t } from "react/jsx-runtime";
import { Variable as s, $ as e, expression as o, ActionValue as i, PlaceholderValue as a } from "@webstudio-is/template";
const n = new s("formState", "initial"), w = {
  category: "data",
  order: 1,
  description: "Collect user data and send it to any webhook.",
  template: /* @__PURE__ */ r(
    e.Form,
    {
      state: o`${n}`,
      onStateChange: new i(["state"], o`${n} = state`),
      children: [
        /* @__PURE__ */ r(
          e.Box,
          {
            "ws:label": "Form Content",
            "ws:show": o`${n} === 'initial' || ${n} === 'error'`,
            children: [
              /* @__PURE__ */ t(e.Label, { children: new a("Name") }),
              /* @__PURE__ */ t(e.Input, { name: "name" }),
              /* @__PURE__ */ t(e.Label, { children: new a("Email") }),
              /* @__PURE__ */ t(e.Input, { name: "email" }),
              /* @__PURE__ */ t(e.Button, { children: new a("Submit") })
            ]
          }
        ),
        /* @__PURE__ */ t(
          e.Box,
          {
            "ws:label": "Success Message",
            "ws:show": o`${n} === 'success'`,
            children: new a("Thank you for getting in touch!")
          }
        ),
        /* @__PURE__ */ t(
          e.Box,
          {
            "ws:label": "Error Message",
            "ws:show": o`${n} === 'error'`,
            children: new a("Sorry, something went wrong.")
          }
        )
      ]
    }
  )
};
export {
  w as meta
};
